habitat_baselines/rl/ddppo/policy/resnet_policy.py

```diff
@@ -353,14 +353,32 @@ def forward(
             goal_observations = observations[
                 IntegratedPointGoalGPSAndCompassSensor.cls_uuid
             ]
-            goal_observations = torch.stack(
-                [
-                    goal_observations[:, 0],
-                    torch.cos(-goal_observations[:, 1]),
-                    torch.sin(-goal_observations[:, 1]),
-                ],
-                -1,
-            )
+            if goal_observations.shape[1] == 2:
+                # Polar Dimensionality 2
+                goal_observations = torch.stack(
+                    [
+                        goal_observations[:, 0],
+                        torch.cos(-goal_observations[:, 1]),
+                        torch.sin(-goal_observations[:, 1]),
+                    ],
+                    -1,
+                )
+            else:
+                assert (
+                    goal_observations.shape[1] == 3
+                ), "Unsupported dimensionality"
+
+                vertical_angle = torch.sin(goal_observations[:, 2])
+                # Polar Dimensionality 3
```

Comment:
That would be great to drop the comment why we multiply horizontal angle sin/cos with vertical sin.